### PR TITLE
Fix a11y tab traversal

### PR DESCRIPTION
**Pull request by yjbanov**
Fix widget traversal when using the TAB key while a11y is enabled.

- Use `editableTextId` (added in https://github.com/flutter/flutter/pull/79683) to synchronize the `flutter/textinput` channel with the semantics tree by queuing commands until an `<input>` element is available.
- Add `tabindex=0` onto tappable targets, otherwise the browser doesn't treat the element as focusable.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: lib/ui/semantics.dart [hunk 1]
```diff
@@ -745,7 +745,12 @@ class SemanticsUpdateBuilder extends NativeFieldWrapperClass2 {
     required Int32List childrenInTraversalOrder,
     required Int32List childrenInHitTestOrder,
     required Int32List additionalActions,
+    int? editableTextId,
   }) {
+    assert(
+      SemanticsFlag.isTextField.index & flags == 0 || editableTextId != null,
+      'Text field semantics node must provide a non-ull editableTextId',
```
Comment (goderbauer) [inline lib/ui/semantics.dart:752]: non-ull -> non-null

File: lib/web_ui/dev/felt_incremental.bat [hunk 2]
```diff
@@ -0,0 +1,70 @@
+:: felt_windows: a command-line utility for Windows for building and testing
```
Comment (goderbauer) [inline lib/web_ui/dev/felt_incremental.bat:1]: Are these changes supposed to be in this PR? They seem unrelated?

Comment (goderbauer) [reply lib/web_ui/dev/felt_incremental.bat:1]: Some of the other files as well...

**Review by goderbauer — COMMENTED** [review]
(haven't looked at the web_UI code in detail since I am not too familiar with that one...)

File: lib/web_ui/dev/felt_incremental.bat (hunk 2 above)
Comment (yjbanov) [reply lib/web_ui/dev/felt_incremental.bat:1]: Yeah, feel free to ignore. I will split this up into multiple PRs. Just sent this draft hastily as I was rushing out the door for vacation :-)

**Comment by yjbanov** [comment]
Closing this in favor of https://github.com/flutter/engine/pull/25797. The new PR does not require framework-side changes. Will be ready for review probably later today.
